spec/light-client/Blockchain.tla

```diff
@@ -0,0 +1,212 @@
+----------------------------- MODULE Blockchain -----------------------------
+(* This is a high-level specification of Tendermint blockchain
+   that is designed specifically for:
+   
+   (1) Lite client, and
+   (2) Fork accountability.
+ *)
+EXTENDS Integers, Sequences
+
+Min(a, b) == IF a < b THEN a ELSE b
+
+CONSTANT
+  AllNodes,
+    (* a set of all nodes that can act as validators (correct and faulty) *)
+  ULTIMATE_HEIGHT,
+    (* a maximal height that can be ever reached (modelling artifact) *)
+  MAX_POWER
+    (* a maximal voting power of a single node *)
+
+Heights == 0..ULTIMATE_HEIGHT   (* possible heights *)
+
+Powers == 1..MAX_POWER          (* possible voting powers *)
+
+(* A commit is just a set of nodes who have committed the block *)
+Commits == SUBSET AllNodes
+
+(* The set of all block headers that can be on the blockchain.
+   This is a simplified version of the Block data structure in the actual implementation. *)
+BlockHeaders == [
+  height: Heights,
+    \* the block height
+  lastCommit: Commits,
+    \* the nodes who have voted on the previous block, the set itself instead of a hash
+  (* in the implementation, only the hashes of V and NextV are stored in a block,
+     as V and NextV are stored in the application state *) 
+  VP: UNION {Nodes \in SUBSET AllNodes \ {{}}: [Nodes -> Powers]},
```

Comment:
Why not VS (ValidatorsSet)?

Reply:
VP and NextVP are functions that assign voting power to every element in their domains. We can see them as multi-sets, but technically they are functions.

Reply:
Can we define this function separately, like we do with Commits for example? It can be called ValidatorSet.

Reply:
Then we would have to carry the function in every header separately, that is, we would have `VS`, `NextVS`, `powerFun`, and `nextPowerFun`. I guess, the voting powers can also change, right? This would clutter the spec.

Reply:
I have introduced two operators: `VS(h)` and `NextVS(h)` return the validator set and the next validator set per header respectively.